server/implementation-servlet/src/main/java/io/smallrye/graphql/entry/http/IndexInitializer.java

```diff
@@ -87,6 +89,7 @@ private IndexView createCustomIndex() {
             indexer.index(convertClassToInputStream(Repeatable.class));
 
             // directives from the API module
+            indexer.index(convertClassToInputStream(Authenticated.class));
```

Comment:
Note mostly to myself: the same change will also be needed in Quarkus (https://github.com/quarkusio/quarkus/blob/3.7.0.CR1/extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallRyeGraphQLProcessor.java#L275)